lib/sanitize.js

```diff
@@ -1,6 +1,24 @@
+var process = require('child_process')
+var command = 'git config --get core.commentchar'
+var comment = '#'
+
+try {
+  // execSync leaks a newline
+  comment = process.execSync(command).toString().trim()
+} catch (error) {
+  if (error.status === 1) {
+    // a) 'core.commentchar' has not been set (git defaults it to #)
+    // b) this is not a git repository
+    // c) maybe something else?
+    // suppress error, use '#'
+  } else {
+    throw (error)
+  }
+}
+
 var scissor = /# ------------------------ >8 ------------------------[\s\S]+/
 module.exports = function (value) {
   return value.replace(scissor, '').split(/\n/).filter(function (line) {
-    return !/^#/.test(line)
+    return !line.match(new RegExp('^' + comment))
```

Comment:
Should never compile regexes inside loop-like structures (here a `filter` call). Move that up a scope:
```
const lineTest = new RegExp(...)
module.exports = ...
  ...
    return lineTest.test(line)
  ...
```
Also use `test` when only interested in a boolean result, not matching individual parts.